File: demo/app/android/app/src/main/kotlin/walletsdk/flutter/converters/fromFlutterTypes.kt
/*
Copyright Gen Digital Inc. All Rights Reserved.

SPDX-License-Identifier: Apache-2.0
*/

package walletsdk.flutter.converters

import dev.trustbloc.wallet.sdk.verifiable.CredentialsArray
import dev.trustbloc.wallet.sdk.verifiable.Opts
import dev.trustbloc.wallet.sdk.verifiable.Verifiable

fun convertToVerifiableCredentialsArray(credentials: List<String>): CredentialsArray {
    val credArray = CredentialsArray()
    for (cred in credentials) {
        val opts = Opts()
        opts.disableProofCheck()

        val parsedCred = Verifiable.parseCredential(cred, opts)
        credArray.add(parsedCred)
    }

    return credArray
}

File: demo/app/ios/Runner/coverters/fromFlutterTypes.swift
/*
Copyright Gen Digital Inc. All Rights Reserved.

SPDX-License-Identifier: Apache-2.0
*/

import Foundation
import Walletsdk

/* Need to be removed */
func convertToVerifiableCredentialsArray(credentials: Array<String>) -> VerifiableCredentialsArray {
    let opts = VerifiableNewOpts()
    opts!.disableProofCheck()
    
    let parsedCredentials = VerifiableCredentialsArray()!
    
    for cred in credentials{
        let parsedVC = VerifiableParseCredential(cred, opts, nil)!
        parsedCredentials.add(parsedVC)
    }

    return parsedCredentials
}

func convertToVerifiableCredentialsArrayV2(credentials: Array<String>, configIds: Array<String>) -> VerifiableCredentialsArrayV2 {
    let opts = VerifiableNewOpts()
    opts!.disableProofCheck()
    
    let parsedCredentials = VerifiableCredentialsArrayV2()!
        
    for i in 0..<credentials.count {
        let parsedVC = VerifiableParseCredential(credentials[i], opts, nil)!
        parsedCredentials.add(parsedVC, configID: configIds[0])
    }
    
    return parsedCredentials
}
